core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -19,138 +19,357 @@
 
 package org.elasticsearch.index.query;
 
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.BooleanClause.Occur;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.lucene.search.Queries;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.query.SimpleQueryParser.Settings;
```

Comment:
:-D You had a good point with the diff being harder to read though (and found a few issues after the diff itself was easier to read). (Yeah, I try to be an optimist here.)